x-pack/plugins/alerting/server/invalidate_pending_api_keys/task.ts

```diff
@@ -29,6 +30,22 @@ import { InvalidatePendingApiKey } from '../types';
 const TASK_TYPE = 'alerts_invalidate_api_keys';
 export const TASK_ID = `Alerts-${TASK_TYPE}`;
 
+const stateSchemaByVersion = {
+  1: {
+    up: (state: Record<string, unknown>) => ({
+      runs: state.runs || 0,
+      total_invalidated: state.total_invalidated || 0,
+    }),
+    schema: schema.object({
+      runs: schema.number(),
+      total_invalidated: schema.number(),
+    }),
+  },
+};
+
+const latestSchema = stateSchemaByVersion[1].schema;
```

Comment:
>Is this something we'll have to manually increment every time we add a new schema?

In this approach, yeah. It would have to be manually incremented every time a new schema is added.

>Wondering if there's anything we can do to automate this>

I tried to see if there was an automated way to get the max key of an object and have that as the type definition but turned unsuccessful. We could explore a CI job for this but the framework would have to find a way to know where / what type definition the task type is using to detect if it's out of date.

I've been thinking maybe this is fine given the developer won't be able to to use the new schema until this value is incremented because the rule executor is bound to `LatestTaskStateSchema` which is derived from `latestSchema` in this example. I'm open to ideas!

Reply:
Maybe we can add the latest schema as latest in the `stateSchemaByVersion` object and bump the version number for the previous latest?
like:

```
const stateSchemaByVersion = {
  latest: {
    up: (...),
    schema: schema.object(...),
  },
};
```

becomes like below when there is a new version: 

```
const stateSchemaByVersion = {
  latest: {
    up: (...),
    schema: schema.object(..., newField:schema.staring()),
  },
 1: {
    up: (...),
    schema: schema.object(...),
  },
};
```

Reply:
Do you think creating `latest` would make a developer think they can modify the schema inline and forget more easily to create `1`, `2`, `3` as they modify their schema?

Reply:
Yes, it's more explicit and changing the schema is enough, no need to update `const latestSchema = stateSchemaByVersion[1].schema;` either.

Reply:
I think we should just use numbers, and not use `latest`. This mirrors the approach that the Kibana Core team is taking with saved-object model versions: https://github.com/elastic/kibana/blob/9b330e493216e8dde3166451e4714966f63f5ab7/packages/core/saved-objects/core-saved-objects-server/src/saved_objects_type.ts#L179-L230

Reply:
Thanks for the input. I think for consistency's sake, it would be good to keep things the same and follow what the core team is doing in this scenario.